server/http/custom_routes.go

```diff
@@ -42,7 +44,8 @@ func (c *customController) Routes() openapi.Routes {
 
 func (c *customController) instrumentRoute(name string, route string, f http.HandlerFunc) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		ctx, span := c.tracer.Start(r.Context(), name)
+		ctx := otel.GetTextMapPropagator().Extract(r.Context(), propagation.HeaderCarrier(r.Header))
```

Comment:
This will use the tracing headers to define the parent span ID. This allows tracetest to be tested by tracetest.